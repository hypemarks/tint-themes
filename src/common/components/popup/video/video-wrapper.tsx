import React, { FC } from 'react';
import { FacebookPlayer } from './player/facebook/facebook-player';
import { VideoPlayer } from './player/other/video-player';
import { VimeoPlayer } from './player/vimeo/vimeo-player';
import { VinePlayer } from './player/vine/vine-player';
import { YoutubePlayer } from './player/youtube/youtube-player';
import { globalConfig } from '../../../config/config';
import { TikTokPlayer } from './player/tiktok/tiktok-player';

interface IVideoWrapperProps {
  videoUrl: string;
  onEnded?: () => void;
  muted?: boolean;
  isMounted: boolean;
}

export const VideoWrapper: FC<IVideoWrapperProps> = ({ videoUrl, onEnded, muted = false, isMounted }) => {
  const youtubeRegExp = globalConfig.regExp.youtubeUrl;
  const facebookRegExp = globalConfig.regExp.facebookUrl;
  const vimeoRegExp = globalConfig.regExp.vimeoUrl;
  const vine = globalConfig.regExp.vineUrl;
  const tiktok = globalConfig.regExp.tikTokClass;

  if (!isMounted) {
    return null;
  }

  if (youtubeRegExp.test(videoUrl)) {
    return <YoutubePlayer videoUrl={videoUrl} onEnded={onEnded} />;
  } else if (facebookRegExp.test(videoUrl)) {
    return <FacebookPlayer videoUrl={videoUrl} onEnded={onEnded} />;
  } else if (vimeoRegExp.test(videoUrl)) {
    return <VimeoPlayer videoUrl={videoUrl} onEnded={onEnded} />;
  } else if (vine.test(videoUrl)) {
    return <VinePlayer videoUrl={videoUrl} />;
  } else if (tiktok.test(videoUrl)) {
    return <TikTokPlayer videoUrl={videoUrl} />;
  }
  return <VideoPlayer videoUrl={videoUrl} onEnded={onEnded} muted={muted} />;
};
