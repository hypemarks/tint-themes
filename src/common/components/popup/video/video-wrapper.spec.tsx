import { render } from '@testing-library/react';
import React from 'react';
import { VideoWrapper } from './video-wrapper';

describe('Test <VideoWrapper /> Component', () => {
  it('should render snapshot', () => {
    const wrapper = render(<VideoWrapper videoUrl={'url'} onEnded={jest.fn} isMounted={true} />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  it('should render video component for youtube', () => {
    const wrapper = render(
      <VideoWrapper
        videoUrl={'https://www.youtube.com/watch?v=3NQnnhlRWmc&ab_channel=PolskieNagrania'}
        onEnded={jest.fn}
        isMounted={true}
      />
    );
    const video = wrapper.queryByTestId('youtube-player');
    expect(video).toBeInTheDocument();
  });

  it('should render video component for facebook', () => {
    const wrapper = render(
      <VideoWrapper videoUrl={'https://www.facebook.com/videos/video'} onEnded={jest.fn} isMounted={true} />
    );
    const video = wrapper.queryByTestId('facebook-player');
    expect(video).toBeInTheDocument();
  });

  it('should render video component for vine', () => {
    const wrapper = render(
      <VideoWrapper videoUrl={'https://www.vine.co/videoaa'} onEnded={jest.fn} isMounted={true} />
    );
    const video = wrapper.queryByTestId('vine-player');
    expect(video).toBeInTheDocument();
  });

  it('should render video component for tiktok', () => {
    const wrapper = render(<VideoWrapper videoUrl={'tiktok-embed'} onEnded={jest.fn} isMounted={true} />);
    const video = wrapper.queryByTestId('tiktok-player');
    expect(video).toBeInTheDocument();
  });

  it('should render video component for vimeo', () => {
    const wrapper = render(<VideoWrapper videoUrl={'https://vimeo.com/76979871'} onEnded={jest.fn} isMounted={true} />);
    const video = wrapper.queryByTestId('vimeo-player');
    expect(video).toBeInTheDocument();
  });

  it('should render video component for other type of social media', () => {
    const wrapper = render(<VideoWrapper videoUrl={'random-video-url'} onEnded={jest.fn} isMounted={true} />);
    const video = wrapper.queryByTestId('video-player');
    expect(video).toBeInTheDocument();
  });
});
