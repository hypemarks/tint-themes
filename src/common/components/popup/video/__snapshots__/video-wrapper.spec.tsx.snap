// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <VideoWrapper /> Component should render snapshot 1`] = `
<DocumentFragment>
  <video
    autoplay=""
    data-testid="video-player"
    height="100%"
    playsinline=""
    preload="auto"
    width="100%"
  >
    <source
      src="url"
      type="video/mp4"
    />
  </video>
</DocumentFragment>
`;
